frontend/rust-lib/flowy-database2/src/services/database_view/view_editor.rs

```diff
@@ -391,8 +391,41 @@ impl DatabaseViewEditor {
     Ok(())
   }
 
-  pub async fn v_delete_group(&self, _params: DeleteGroupParams) -> FlowyResult<()> {
-    Ok(())
+  pub async fn v_delete_group(&self, group_id: &str) -> FlowyResult<RowsChangePB> {
```

Comment:
```rust
pub async fn v_delete_group(&self, group_id: &str) -> FlowyResult<RowsChangePB> {
    let group_controller = self.group_controller.write().await;
    let controller = match group_controller.as_mut() {
        Some(controller) => controller,
        None => return Ok(RowsChangePB::default()),
    };

    let old_field = self.delegate.get_field(controller.field_id());
    let (row_ids, type_option_data) = controller.delete_group(group_id)?;

    let mut changes = RowsChangePB::default();

    if let Some(field) = old_field {
        let deleted_rows = row_ids
            .iter()
            .filter_map(|row_id| self.delegate.remove_row(row_id))
            .map(|row| row.id.into_inner());

        changes.deleted_rows.extend(deleted_rows);

        if let Some(type_option) = type_option_data {
            self.delegate
                .update_field(&self.view_id, type_option, field)
                .await?;
        }

        let notification = GroupChangesPB {
            view_id: self.view_id.clone(),
            deleted_groups: vec![group_id.to_string()],
            ..Default::default()
        };
        notify_did_update_num_of_groups(&self.view_id, notification).await;
    }

    Ok(changes)
}

```
1. reduce nesting
2. no need to use tuple
3. return asap if the group controller is None